ql/instruments/makecms.cpp

```diff
@@ -201,8 +201,8 @@ namespace QuantLib {
 
     MakeCms& MakeCms::withDiscountingTermStructure(
                 const Handle<YieldTermStructure>& discountingTermStructure) {
-        engine_ = boost::shared_ptr<PricingEngine>(new
-                            DiscountingSwapEngine(discountingTermStructure));
+        engine_ = boost::static_pointer_cast<PricingEngine>(
```

Comment:
As engine_ is a boost::shared_ptr<PricingEngine> static_pointer_cast make explicit an  implicit conversion with no run-time impact. I usually use the cast when I know the code must be compiled on different OS , because some compiler may complain about it.